client/src/components/ProductCard/ProductCard.js

```diff
@@ -1,9 +1,20 @@
-import { Card } from "react-bootstrap";
-// import { BsHeart, BsHeartFill } from 'react-icons/bs';
+import { Card, OverlayTrigger, Tooltip } from "react-bootstrap";
 import { Link } from "react-router-dom";
 import Moment from "react-moment";
+import React, { useRef } from "react";
+
+function Date({ children }) {
+  return <Moment format="MMM D YYYY HH:mm a">{children}</Moment>;
+}
 
 function ProductCard({ params }) {
+  const locationRef = useRef(null);
+  const dateRef = useRef(null);
+  const locationIsOverflowing =
```

Comment:
This handles detection of the ellipsis based on the offset and scroll width. When scroll width exceeds offset, we pass a false flag into the OverflowTrigger prop to turn it off. 

Note: this doesn't account for responsive design so this would need a window resize observer.